scripts/simplify_network.py

```diff
@@ -522,6 +523,91 @@ def consense(x):
     return clustering.network, clustering.busmap
 
 
+def drop_isolated_nodes(n, drop_e_threshold, n_periods_in_year=8760):
+    generators_sum_origin = n.generators.p_nom.sum()
+    load_sum_origin = n.loads_t.p_set.sum().sum()
+
+    loads_t_ann = n_periods_in_year * n.loads_t.p_set.mean(axis=0)
+
+    # duplicated sub-networks mean that there is at least one interconnection between buses
+    i_islands = n.buses[
+        (~n.buses.duplicated(subset=["sub_network"], keep=False))
+        & (n.buses.carrier == "AC")
+    ].index
+    i_islands_drop = i_islands[loads_t_ann[i_islands] <= drop_e_threshold]
+
+    i_generators_drop = n.generators[n.generators.bus.isin(i_islands_drop)].index
+
+    n.mremove("Bus", i_islands_drop)
+    n.mremove("Load", i_islands_drop)
+    n.mremove("Generator", i_generators_drop)
+
+    load_sum_final = n.loads_t.p_set.sum().sum()
+    generators_sum_final = n.generators.p_nom.sum()
+
+    logger.info(
+        f"Dropped {len(i_islands)} buses. A resulted load discrepancy is {(100 * ((load_sum_final - load_sum_origin)/load_sum_origin)):2.1E}% and {(100 * ((generators_sum_final - generators_sum_origin)/generators_sum_origin)):2.1E}% for average load and generation capacity, respectivelly"
+    )
+
+    return n
+
+
+def merge_isolated_buses(n, drop_p_threshold=0, n_periods_in_year=8760):
```

Comment:
The core of this function may be modified similarly to aggregate_to_substations

That function is quite close to what we would like to achieve with this merge isolated buses